### PR TITLE
Shell refactor

### PR DESCRIPTION
**Pull request by objmagic**
Refactor handlers part so that twitter internal heron-shell can use most of the codes from the OSS side.

### CODE REVIEW

**Comment by billonahill** [comment]
@objmagic could you please provide a description explaining the motivation for the refactor?

File: heron/shell/src/python/main.py [hunk 1]
```diff
@@ -21,25 +21,9 @@
...
-    (r"^/file/(.*)", handlers.FileHandler),
-    (r"^/filedata/(.*)", handlers.FileDataHandler),
-    (r"^/filestats/(.*)", handlers.FileStatsHandler),
-    (r"^/download/(.*)", handlers.DownloadHandler),
-    (r"^/killexecutor", handlers.KillExecutorHandler),
-])
-
-
-if __name__ == '__main__':
+def run(extra_handlers=[]):
```
Comment (billonahill) [inline heron/shell/src/python/main.py]: Could we instead allow the caller to pass the entire list of handlers, so they could either remove or add to the defaults?
```
def run(handlers=utils.default_url_to_handlers)
```

Comment (objmagic) [inline heron/shell/src/python/main.py]: i see little possibility that we will need to remove in the future. And if we want to enable remove here, we may want to switch to use dict. find and remove a tuple in a list is quite awkward.

**Comment by huijunw** [comment]
shipit after ci pass

File: heron/shell/src/python/main.py (hunk 1 above)
Comment (billonahill) [inline heron/shell/src/python/main.py]: I wouldn't expect someone to remove from the url_to_handlers list, but instead pass a new list, that might include some of those entries.

**Pull request merged**